app/services/PaymentServices.scala

```diff
@@ -1,78 +1,67 @@
 import com.typesafe.config.Config
 import data.ContributionData
+import models.PaymentMode
+import models.PaymentMode.{Default, Testing}
 import play.api.mvc.RequestHeader
-import services.PaymentServices.{Default, Mode, Testing}
 
 import scala.concurrent.ExecutionContext
-import scala.concurrent.ExecutionContext.Implicits.global
 
-
-object PaymentServices {
-
-  /* https://stripe.com/docs/dashboard#livemode-and-testing
-   */
-  sealed trait Mode {
-    val name: String
-  }
-
-  case object Default extends Mode {
-    val name = "default"
-  }
-
-  case object Testing extends Mode {
-    val name = "testing"
-  }
-
-  object Mode {
-    val all: Set[Mode] = Set(Default, Testing)
-  }
-
-  def stripeServiceFor(stripeConfig: Config, universe: Mode, contributionData: ContributionData): StripeService = {
-    val stripeMode = stripeConfig.getString(universe.name)
-    val keys = stripeConfig.getConfig(s"keys.$stripeMode")
-    val credentials = StripeCredentials(
-      secretKey = keys.getString("secret"),
-      publicKey = keys.getString("public")
-    )
-
-    val metrics = new ServiceMetrics(stripeMode, "giraffe","stripe")
-    new StripeService(StripeApiConfig(stripeMode, credentials), metrics, contributionData)
-  }
-
-  def stripeServicesFor(stripeConfig: Config, contributionDataPerMode: Map[Mode, ContributionData]):Map[Mode, StripeService] = {
-    Mode.all.map(mode => mode -> stripeServiceFor(stripeConfig, mode, contributionDataPerMode(mode))).toMap
+class PaymentServices(
+  config: Config,
+  authProvider: AuthenticatedIdUser.Provider,
+  testUsernames: TestUsernames,
+  identityService: IdentityService,
+  contributionDataPerMode: Map[PaymentMode, ContributionData],
+  actorSystem: ActorSystem
+)(implicit ec: ExecutionContext) {
+
+  val stripeServices: Map[PaymentMode, StripeService] = {
+    val stripeConfig = config.getConfig("stripe")
+
+    def stripeServiceFor(mode: PaymentMode): StripeService = {
+      val contributionData = contributionDataPerMode(mode)
+      val stripeMode = stripeConfig.getString(mode.name)
+      val keys = stripeConfig.getConfig(s"keys.$stripeMode")
+      val credentials = StripeCredentials(
+        secretKey = keys.getString("secret"),
+        publicKey = keys.getString("public")
+      )
+      val metrics = new ServiceMetrics(stripeMode, "giraffe", "stripe")
+      new StripeService(
+        apiConfig = StripeApiConfig(stripeMode, credentials),
+        metrics = metrics,
+        contributionData = contributionData,
+        identityService = identityService
+      )
+    }
+    PaymentMode.all.map(mode => mode -> stripeServiceFor(mode)).toMap
   }
 
-  def paypalServicesFor(paypalConfig: Config, contributionDataPerMode: Map[Mode, ContributionData])(ec: ExecutionContext): Map[Mode, PaypalService] = {
+  val paypalServices: Map[PaymentMode, PaypalService] = {
+    val paypalExecutionContext = actorSystem.dispatchers.lookup("contexts.paypal-context")
```

Comment:
Ah, I've just always manually created a separate `ExecutionContext` in this kind of case, as I'm allergic to config files.